File: standard-udfs-examples/standard-udfs-example-udfs-avro/src/test/java/com/linkedin/stdudfs/examples/avro/TestStructCreateByIndexFunctionWrapper.java
/**
 * Copyright 2018 LinkedIn Corporation. All rights reserved.
 * Licensed under the BSD-2 Clause license.
 * See LICENSE in the project root for license information.
 */
package com.linkedin.stdudfs.examples.avro;

import com.google.common.collect.ImmutableList;
import org.apache.avro.Schema;
import org.testng.annotations.Test;

import static com.linkedin.stdudfs.avro.common.AssertAvroUdf.*;
import static com.linkedin.stdudfs.avro.common.SchemaFactory.*;


public class TestStructCreateByIndexFunctionWrapper {
  @Test
  public void testStructCreateByIndex() {
    assertFunction(new StructCreateByIndexFunctionWrapper(), new Schema[]{STRING, STRING}, new Object[]{"x", "y"},
        ImmutableList.of("x", "y"));

    assertFunction(new StructCreateByIndexFunctionWrapper(), new Schema[]{STRING, STRING}, new Object[]{null, "y"},
        null);

    assertFunction(new StructCreateByIndexFunctionWrapper(), new Schema[]{STRING, STRING}, new Object[]{"x", null},
        null);
  }
}

File: transportable-udfs-examples/transportable-udfs-example-udfs-spark/src/test/scala/com/linkedin/transport/examples/spark/TestStructCreateByIndexFunctionWrapper.scala
/**
 * Copyright 2018 LinkedIn Corporation. All rights reserved.
 * Licensed under the BSD-2 Clause license.
 * See LICENSE in the project root for license information.
 */
package com.linkedin.transport.examples.spark

import com.linkedin.transport.spark.common.AssertSparkExpression._
import org.apache.spark.sql.Row
import org.testng.annotations.{BeforeClass, Test}

import scala.collection.mutable.WrappedArray._

class TestStructCreateByIndexFunctionWrapper {

  @BeforeClass
  def registerFunction(): Unit = {
    registerStandardUdf(
      "struct_create_by_index",
      classOf[StructCreateByIndexFunctionWrapper]
    )
  }

  @Test
  def testStructCreateByIndexFunction(): Unit = {
    assertFunction("struct_create_by_index('x', 'y')", Row("x", "y"))
    assertFunction("struct_create_by_index(1, array(1))", Row(1, make(Array(1))))
    assertFunction("struct_create_by_index(null, null)", null)
  }
}
